modules/api/src/test/java/org/apache/fluo/api/client/scanner/ScannerStreamTest.java

```diff
@@ -0,0 +1,219 @@
+
+    CellScannerImpl cellScanStream = new CellScannerImpl(rowCols);
+
+    rowCols.stream().filter(rcv -> rcv.getColumn().getsFamily().equals("f2"))
+        .collect(Collectors.toSet());
+    cellScanStream.stream().filter(rcv -> rcv.getColumn().getsFamily().equals("f2"))
+        .collect(Collectors.toSet());
+
+    Assert.assertNotEquals(empty, cellScanStream.stream().collect(Collectors.toSet()));
+    Assert.assertEquals(rowCols, cellScanStream.stream().collect(Collectors.toSet()));
+    Assert.assertEquals(rowCols.size(), cellScanStream.stream().count());
+  }
+
+  @Test
+  public void testColumnScannerStream() {
+    Set<ColumnValue> colsVal = new HashSet<>();
+    Set<ColumnValue> empty = new HashSet<>();
+
+    Bytes row = Bytes.of("123");
+    colsVal.add(new ColumnValue(new Column("f1", "q1"), Bytes.of("v1")));
+    colsVal.add(new ColumnValue(new Column("f2", "q3"), Bytes.of("v2")));
+    colsVal.add(new ColumnValue(new Column("f1", "q1"), Bytes.of("v3")));
+    colsVal.add(new ColumnValue(new Column("f1", "q2"), Bytes.of("v4")));
+    colsVal.add(new ColumnValue(new Column("f2", "q5"), Bytes.of("v5")));
+
+    ColumnScanner colScanStream = new ColumnScannerImpl(row, colsVal);
+
+    colsVal.stream().filter(cv -> cv.getColumn().getsFamily().equals("f2"))
+        .collect(Collectors.toSet());
+    colScanStream.stream().filter(cv -> cv.getColumn().getsFamily().equals("f2"))
+        .collect(Collectors.toSet());
+
+    Assert.assertNotEquals(empty, colScanStream.stream().collect(Collectors.toSet()));
+    Assert.assertEquals(colsVal, colScanStream.stream().collect(Collectors.toSet()));
+    Assert.assertEquals(colsVal.size(), colScanStream.stream().count());
+  }
+
+  @Test
+  public void testRowScannerStream() {
+    List<ColumnScanner> rows = new ArrayList<>();
+    Set<ColumnValue> cv1 = new HashSet<>();
+    Set<ColumnValue> cv2 = new HashSet<>();
+
+    Bytes row1 = Bytes.of("555555555");
+    cv1.add(new ColumnValue(new Column("firstname"), Bytes.of("Chris")));
+    cv1.add(new ColumnValue(new Column("lastname"), Bytes.of("McTague")));
+    cv1.add(new ColumnValue(new Column("age"), Bytes.of("21")));
+
+    rows.add(new ColumnScannerImpl(row1, cv1));
+
+    Bytes row2 = Bytes.of("55234234");
+    cv2.add(new ColumnValue(new Column("firstname"), Bytes.of("Hulk")));
+    cv2.add(new ColumnValue(new Column("lastname"), Bytes.of("Hogan")));
+    cv2.add(new ColumnValue(new Column("age"), Bytes.of("60")));
+
+    rows.add(new ColumnScannerImpl(row2, cv2));
+
+    RowScannerImpl rsi = new RowScannerImpl(rows);
+
+    Set<Person> people = rsi.stream().map(cs -> toPerson(cs)).collect(Collectors.toSet());
+
+    Set<Person> expected = new HashSet<>();
+    expected.add(new Person("Chris", "McTague", 21, 555555555));
+    expected.add(new Person("Hulk", "Hogan", 60, 55234234));
+
+    Assert.assertEquals(expected.hashCode(), people.hashCode());
+    Assert.assertEquals(expected.size(), people.size());
```

Comment:
After adding a equals method to the Person type , you can replace these two checks with :

```
  Assert.assertEquals(expected, people);
```